glean-core/src/database/mod.rs

```diff
@@ -707,4 +808,41 @@ mod test {
             );
         }
     }
+
+    #[test]
+    fn test_deferred_ping_collection() {
+        // Init the database in a temporary directory.
+        let dir = tempdir().unwrap();
+        let str_dir = dir.path().display().to_string();
+        let db = Database::new(&str_dir, true).unwrap();
+
+        assert!(db.ping_lifetime_data.is_some());
+
+        // Attempt to record a known value.
+        let test_value = "test-value";
+        let test_storage = "test-storage1";
+        let test_metric_id = "telemetry_test.test_name";
+        db.record_per_lifetime(
+            Lifetime::Ping,
+            test_storage,
+            test_metric_id,
+            &Metric::String(test_value.to_string()),
+        )
+        .unwrap();
+
+        // Verify that the data is correctly recorded.
+        let mut found_metrics = 0;
+        let mut snapshotter = |metric_name: &[u8], metric: &Metric| {
+            found_metrics += 1;
+            let metric_id = String::from_utf8_lossy(metric_name).into_owned();
+            assert_eq!(test_metric_id, metric_id);
+            match metric {
+                Metric::String(s) => assert_eq!(test_value, s),
+                _ => panic!("Unexpected data found"),
+            }
+        };
+
+        db.iter_store_from(Lifetime::Ping, test_storage, None, &mut snapshotter);
```

Comment:
One thing you should probably do is manually check if data was written to rkv. You might want to do something [like this](https://github.com/mozilla/glean/blob/15bd43b316e3829bd4d747923945f209ad9bb72c/glean-core/src/database/mod.rs#L214-L220).

Reply:
I added code related to this, I am not sure it works 🤔

Reply:
What makes you think this is not working?